src/pal/pal_posix.h

```diff
@@ -264,5 +270,31 @@ namespace snmalloc
 
       OS::error("Out of memory");
     }
+
+    /**
+     * Source of Entropy
+     *
+     * This is a default that works on many POSIX platforms.
+     */
+    static uint64_t get_entropy64()
+    {
+      if constexpr (!pal_supports<Entropy, OS>)
+      {
+        // Derived Pal does not provide entropy.
+        return 0;
+      }
+      else if constexpr (OS::get_entropy64 != get_entropy64)
```

Comment:
I'm sure I'm missing something, but doesn't `OS` inherit from us?  That is, if this condition is true we won't be here in the first place?

Reply:
OS could override.  In which case the overriding definition will take over that is
  `OS::get_entropy64`
will be the override, and 
  `get_entropy 64`
will be the non-overriden definition. In my experiments, this did the right thing, but I may be wrong.  And this is not tested by CI, so extra care is required.

Reply:
Sorry, what I mean is: if `Pal` is `OS<Pal_Posix>` then `Pal::get_entropy64` will be `OS::get_entropy64` not `Pal_Posix::get_entropy64`, and so we won't even get to this `if constexpr` to consider it...  unless we refer to `Pal_Posix` directly somewhere (except to extend from it)?

Reply:
I think inside that branch of the `constexpr` we could put anything that compiles on any platform.

This definition is useful, if any of the Pal_Posix layer requires Entropy itself.  Let's say we add our own ASLR to the Pal_Posix, then that can use the system supplied Entropy, which may be overridden by the platform.  Does that make sense?

Reply:
@nwf is this okay with you?